src/products/components/ProductVariantSetDefault/ProductVariantSetDefault.tsx

```diff
@@ -0,0 +1,86 @@
+import CardMenu from "@saleor/components/CardMenu";
+import useNotifier from "@saleor/hooks/useNotifier";
+import { useProductVariantSetDefaultMutation } from "@saleor/products/mutations";
+import { getProductErrorMessage } from "@saleor/utils/errors";
+import React from "react";
+import { useIntl } from "react-intl";
+
+interface SimpleProductVariant {
+  id: string;
+  sku: string;
+  default?: boolean;
+  name: string;
+}
+
+interface ProductVariantSetDefaultProps {
+  variant: SimpleProductVariant;
+  productId?: string;
+}
+
+const ProductVariantSetDefault: React.FC<ProductVariantSetDefaultProps> = ({
+  variant,
+  productId
+}) => {
+  const intl = useIntl();
+  const notify = useNotifier();
+
+  const getProductVariantSetDefaultVariables = (
+    productId: string,
+    variantId: string
+  ) => ({
+    variables: {
+      productId,
+      variantId
+    }
+  });
+
+  const [productVariantSetDefault] = useProductVariantSetDefaultMutation({
```

Comment:
we use mutations and queries in views(views/ProductVariant). onSetDefaultVariant should be passed here as prop.

Reply:
☝️